pkg/statistics/handle/ddl/ddl.go

```diff
@@ -206,6 +156,22 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 		// Change id for global stats, since the data has not changed!
 		// Note: This operation will update all tables related to statistics with the new ID.
 		return h.statsWriter.ChangeGlobalStatsID(oldSingleTableID, globalTableInfo.ID)
+	case model.ActionRemovePartitioning:
+		// Change id for global stats, since the data has not changed!
+		// Note: This operation will update all tables related to statistics with the new ID.
+		oldTblID, newSingleTableInfo, droppedPartInfo := s.GetRemovePartitioningInfo()
+		if err := h.statsWriter.ChangeGlobalStatsID(oldTblID, newSingleTableInfo.ID); err != nil {
+			return err
+		}
+
+		// Remove partition stats.
+		for _, def := range droppedPartInfo.Definitions {
+			if err := h.statsWriter.UpdateStatsMetaVersionForGC(def.ID); err != nil {
+				return err
+			}
+		}
+	case model.ActionFlashbackCluster:
+		return h.statsWriter.UpdateStatsVersion()
```

Comment:
Do we need to add a default branch here?

Reply:
yes, raise an error here